src/osal/lv_freertos.c

```diff
@@ -260,13 +268,26 @@ lv_result_t lv_thread_sync_wait(lv_thread_sync_t * pxCond)
 
 lv_result_t lv_thread_sync_signal(lv_thread_sync_t * pxCond)
 {
-#if USE_FREERTOS_TASK_NOTIFY
-    /* Send a notification to the task waiting. */
-    xTaskNotifyGive(pxCond->xTaskToNotify);
-#else
     /* If the cond is uninitialized, perform initialization. */
     prvCheckCondInit(pxCond);
 
+#if USE_FREERTOS_TASK_NOTIFY
+    _enter_critical();
+    TaskHandle_t task_to_notify = pxCond->xTaskToNotify;
+    pxCond->xTaskToNotify = NULL;
+    if(task_to_notify == NULL) {
+        /* No task waiting to be notified. Send this signal for later */
+        pxCond->xSyncSignal = pdTRUE;
+    }
+    /* If a task is already waiting, there is no need to set the sync signal */
+    _exit_critical();
+
```

Comment:
Same here, you exit from critical section but still rely on the `task_to_notify`. Not thread safe.